uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/dom/single/impl/BaseSingletonDOMElementFactory.java

```diff
@@ -72,6 +76,28 @@ public void execute() {
         });
     }
 
+    @Override
+    public E createDomElement(final GridLayer gridLayer,
+                              final GridWidget gridWidget,
+                              final GridBodyCellRenderContext context) {
+        widget = createWidget();
+        e = createDomElementInternal();
+
+        widget.addDomHandler(destroyOrFlushkeyDownHandler(context),
+                             KeyDownEvent.getType());
+        widget.addDomHandler((e) -> e.stopPropagation(),
+                             KeyDownEvent.getType());
+        widget.addDomHandler((e) -> e.stopPropagation(),
+                             MouseDownEvent.getType());
+        widget.addDomHandler((e) ->
+                             {
+                                 flush();
+                                 gridLayer.batch();
+                                 gridPanel.setFocus(true);
+                             }, BlurEvent.getType());
```

Comment:
@jomarko @manstis 
I suspect that if you want to manage the **Blur** event on a widget, you have to be sure the widget is _focusable_. Could you please verify that this code works on a _not-focusable_ widget? If, on the other side, the widget must be _focusable_, then enforce this on the generic declaration

Reply:
@gitgabrio good point, I will add check  `if (widget instanceof Focusable)`.